src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecret.java

```diff
@@ -102,11 +80,31 @@ public TeeTaskComputeSecret(
             String fixedSecretOwner,
             String key,
             String value) {
-        this.onChainObjectType = onChainObjectType;
-        this.onChainObjectAddress = onChainObjectAddress;
-        this.secretOwnerRole = secretOwnerRole;
-        this.fixedSecretOwner = fixedSecretOwner;
-        this.key = key;
+        this.header = new TeeTaskComputeSecretHeader(
+                onChainObjectType,
+                onChainObjectAddress,
+                secretOwnerRole,
+                fixedSecretOwner,
+                key
+        );
         this.value = value;
     }
+
+    public TeeTaskComputeSecret withValue(String newValue) {
+        return new TeeTaskComputeSecret(header, newValue);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Right. According to JPA buddy, using `@EqualsAndHashCode` can cause severe performance and memory consumption issues.

Reply:
Sorry, there is still a point that puzzles me.
If everything above is correct, shouldn't it be applied to `TeeTaskComputeSecretHeader` as well ?
I'm wondering why it could be safe to use the annotation on the embedded id and not on the main entity.

Reply:
You're right. Let's make these similar. Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/142/commits/015438d1d56bd1eef2e8cc8271c88c65dcf49f60.